eland/ml/pytorch/transformers.py

```diff
@@ -619,9 +619,8 @@ def _create_tokenization_config(self) -> NlpTokenizationConfig:
     def _create_config(self) -> NlpTrainedModelConfig:
         tokenization_config = self._create_tokenization_config()
 
-        # Set squad well known defaults
+        # Set span and truncate defaults for question answering
         if self._task_type == "question_answering":
-            tokenization_config.max_sequence_length = 386
             tokenization_config.span = 128
             tokenization_config.truncate = "none"
```

Comment:
Span and sequence length are inextricably linked within the model. It is weird to me that we overwrite one but not the other.

Also, to be sure, have we tested scenarios where `span + max_sequence_length > model_max_input_length`?

Reply:
`max_sequence_length` is fixed forever but `span` is update-able at inference time. Tweaking the `span` setting often changes the top result from a Question Answering model.

`max_sequence_length` comes from the HG code but the HG tokenizers do not use the concept of span, we are not overwriting it here we are introducing it. 

> Also, to be sure, have we tested scenarios where span + max_sequence_length > model_max_input_length?

Yes coverage here: https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/inference/nlp/tokenizers/BertTokenizerTests.java

But I am worried that we do not handle the case where the first sequence (i.e. the question) is longer than `model_max_input_length`